cloud-template/src/main/java/com/sequenceiq/cloudbreak/cloud/template/AbstractResourceConnector.java

```diff
@@ -180,9 +185,29 @@ public ExternalDatabaseStatus getDatabaseServerStatus(AuthenticatedContext authe
         context.addGroupResources(scalingGroup.getName(), groupResourceService.getGroupResources(variant, resources));
 
         //compute
+        diskReattachment(resources, scalingGroup, context);
         return computeResourceService.buildResourcesForUpscale(context, auth, stack, Collections.singletonList(scalingGroup));
     }
 
+    protected void diskReattachment(List<CloudResource> resources, Group scalingGroup, ResourceBuilderContext context) {
+        List<CloudResource> diskSets = resources.stream()
+                .filter(cloudResource -> scalingGroup.getName().equalsIgnoreCase(cloudResource.getGroup()))
+                .filter(cloudResource -> getDisResourceType().equals(cloudResource.getType()))
+                .filter(cloudResource -> StringUtils.isEmpty(cloudResource.getInstanceId()) || CommonStatus.DETACHED.equals(cloudResource.getStatus()))
+                .collect(Collectors.toList());
+        for (int i = 0; i < diskSets.size(); i++) {
+            CloudResource cloudResource = diskSets.get(i);
+            Optional.ofNullable(scalingGroup.getInstances().get(i))
+                    .map(CloudInstance::getTemplate)
+                    .map(InstanceTemplate::getPrivateId)
+                    .ifPresent(privateId -> context.addComputeResources(privateId, List.of(cloudResource)));
+        }
+    }
+
+    protected ResourceType getDisResourceType() {
+        throw new IllegalArgumentException("Please override the disk resource type or implement the provider specified disk reattachment");
```

Comment:
Can you you delegate the responsibilty to implenetors by using an abstract method here?

Reply:
It is mandatory. We need to force them to implement if they want to use the disk reattachment out of the box. As the message says.

Reply:
I understand that, what I meant is people need to remember to implement this. The exception will only come in runtime, however if this would be an abstract method, it would not even compile.

Reply:
Unfortunately, I need to implement in Azure, Yarn and Mock as well if I transform to an abstract method and I can return with null only and in that case, only an NPE will occur. It will happen if some try to implement in these 3 other providers.